apps/src/codebridge/utils/fileUtils.ts

```diff
@@ -35,3 +39,61 @@ export function getFileIconNameAndStyle(file: ProjectFile): {
     return {iconName: 'eye', iconStyle: 'regular'};
   }
 }
+
+/**
+ * Prepare the source for saving in levelbuilder. This moves the validation file
+ * into a separate field and removes it from the files and list of open files, if
+ * it was open.
+ *
+ * We split out the validation file because it is handled differently from start sources
+ * in user code. The validation file is not saved to the user's project, and we always use
+ * the validation file from the level, even if the level is template backed (and we are using
+ * start code from the template).
+ * @param source: MultiFileSource
+ * @returns {parsedSource: MultiFileSource, validationFile: ProjectFile}
+ */
+export function prepareSourceForLevelbuilderSave(source?: MultiFileSource) {
+  if (!source) {
+    return {parsedSource: null, validationFile: null};
+  }
+  const newFiles = Object.fromEntries(
+    Object.entries(source.files).filter(
+      ([_, file]) => file.type !== ProjectFileType.VALIDATION
+    )
+  );
+  let validationFile =
+    Object.values(source.files).find(
+      f => f.type === ProjectFileType.VALIDATION
+    ) || null;
+  let openFiles = source.openFiles;
+  if (validationFile && source.openFiles?.includes(validationFile.id)) {
+    openFiles = source.openFiles.filter(id => id !== validationFile?.id);
+    //validationFile.open = false;
```

Comment:
do you need this?

Reply:
good catch, nope because of the next line